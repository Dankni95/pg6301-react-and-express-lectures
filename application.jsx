import React from "react";

export function Application({who}) {
  return <h1>Hello {who}</h1>;
}
