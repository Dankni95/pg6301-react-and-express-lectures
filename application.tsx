import * as React from "react";

export function Application({ who }: { who: string }) {
  return <h1>Hello {who}</h1>;
}
