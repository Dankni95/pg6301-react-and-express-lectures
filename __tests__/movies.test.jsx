import * as React from "react";
import {render} from "react-dom";
import {ListMovies, NewMovie} from "../application";
import {Simulate} from "react-dom/test-utils";

describe("Movies", () => {
    it("runs shows movies", () => {
        const movies = [
            {
                title: "don't look up", year: 2021, plot: "Global disaster"
            },
            {
                title: "alien", year: 1979, plot: "Horror in space"
            }
        ]
        const element = document.createElement("div");
        render(<ListMovies movies={movies} />, element);
        expect(element.innerHTML).toMatchSnapshot();
    });

    it("creates new movie", () => {
        const element = document.createElement("div");
        const onAddMovie = jest.fn();
        render(<NewMovie onAddMovie={onAddMovie} />, element);
        expect(element.innerHTML).toMatchSnapshot();
        Simulate.change(element.querySelector("input"), {target: {value: "Some movie"}});
        expect(onAddMovie).toBeCalledWith({
            title: "Some movie", year: "", plot: ""
        })
    })
})
