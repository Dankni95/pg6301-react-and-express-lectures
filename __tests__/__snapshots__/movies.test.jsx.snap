// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies runs shows movies 1`] = `"<div><h1>Movies</h1><div><h2>don't look up (2021)</h2><div>Global disaster</div></div><div><h2>alien (1979)</h2><div>Horror in space</div></div></div>"`;
